src/coreclr/vm/codeman.cpp

```diff
@@ -2898,15 +2844,15 @@ void EECodeGenManager::allocCode(MethodDesc* pMD, size_t blockSize, size_t reser
 
     // Scope the lock
     {
-        CrstHolder ch(&m_CodeHeapCritSec);
+        CrstHolder ch(&m_CodeHeapLock);
 
         *ppCodeHeap = NULL;
-        TADDR pCode = (TADDR) allocCodeRaw(&requestInfo, sizeof(TCodeHeader), totalSize, alignment, ppCodeHeap);
+        TADDR pCode = (TADDR) AllocCodeWorker(&requestInfo, sizeof(TCodeHeader), totalSize, alignment, ppCodeHeap);
         _ASSERTE(*ppCodeHeap);
 
         if (pMD->IsLCGMethod())
         {
-            pMD->AsDynamicMethodDesc()->GetLCGMethodResolver()->m_recordCodePointer = (void*) pCode;
+            pMD->AsDynamicMethodDesc()->GetLCGMethodResolver()->RecordCodePointer((void*)pCode);
```

Comment:
Allocate the linked list note for the recording before the call to `AllocCodeWorker` so that `RecordCodePointer` cannot fail and leave a piece of memory that's not attached to anything behind?

Reply:
Sure.